tools/mixxx-isolate.py

```diff
@@ -0,0 +1,496 @@
+    lastCore = -1
+    for line in cpuinfo.splitlines():
+        if not line.strip():
+            continue
+        topic, data = line.split(":")
+        topic = topic.strip()
+        if topic == "processor":
+            currentCpu = int(data)
+        elif topic == "core id":
+            coreId = int(data)
+            cpuToCore[currentCpu] = coreId
+            lastCore = max(lastCore, coreId)
+
+    useCore = -1
+    if userCpu == -1:
+        useCore = lastCore
+    else:
+        useCore = cpuToCore.get(userCpu, -1)
+        if useCore == -1:
+            print("Selected CPU does not exist. Using autodetect.")
+            useCore = lastCore
+
+    useCpu = -1
+    isolate = []
+    system = []
+    for cpu, core in cpuToCore.items():
+        if core == useCore and useCpu == -1:
+            useCpu = cpu
+        elif core == useCore:
+            isolate.append(cpu)
+        else:
+            system.append(cpu)
+
+    return {"use": useCpu, "isolate": isolate, "system": system}
+
+
+def cpusetname(name):
+    if name and name[0] == "/":
+        return name
+    return "/%s" % name
+
+
+def deleteCpuset(name, debug):
+    try:
+        setInstance = cset.unique_set(name)
+    except cset.CpusetNotFound:
+        return True
+    # move processes until empty
+    processes = 1
+    while processes:
+        try:
+            log.debug(" Move processes from cpuset %s %s", setInstance, name)
+            proc.move(setInstance, "/", None, False)
+            processes = setInstance.tasks
+            log.debug(" Processes left: %s", processes)
+            break
+        except BrokenPipeError:
+            continue
+        except Exception as e:
+            log.exception(e)
+            print(e)
+
+    ccset.destroy(name)
+    return True
+
+
+def teardown(args, isolate):
+    cset.CpuSet(None)
+
+    if not args.no_offline:
+        if args.cleanup:
+            # since our detection algorithm can't detect isolation cpus
+            # form offline CPUs, we will just bring up all cores on teardown
+            for fname in os.listdir("/sys/devices/system/cpu"):
+                try:
+                    if re.match("cpu[0-9]+", fname):
+                        with open(
+                            "/sys/devices/system/cpu/%s/online" % fname, "w"
+                        ) as fp:
+                            fp.write("1")
+                except OSError:
+                    pass
+                except Exception as e:
+                    print(e)
+
+        for iso in isolate:
+            with open("/sys/devices/system/cpu/cpu%d/online" % iso, "w") as fp:
+                fp.write("1")
+
+    log.info(" Deactivate cpusets")
+    deleteCpuset(args.cpuset_engine, args.debug)
+    deleteCpuset(args.cpuset_isolation, args.debug)
+    deleteCpuset(args.cpuset_system, args.debug)
+
+    if prev_gov is not None:
+        log.info(" Restore governor %s" % prev_gov)
+        system("sudo cpufreq-set -g %s" % prev_gov)
```

Comment:
This looks quite worrying from a security stand point:
- Can we make sure we don't hardcode a `sudo` here and either expect the user to run the script with `sudo` or provide the right permission setup instruction 
- Can we make sure we don't eval an arbitrary string inside a `system` command